crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1348,16 +1395,7 @@ impl RpcServerConfig {
                             .option_layer(Self::maybe_cors_layer(cors)?)
                             .option_layer(Self::maybe_jwt_layer(self.jwt_secret)),
                     )
-                    .set_rpc_middleware(
-                        RpcServiceBuilder::new().layer(
-                            modules
-                                .http
-                                .as_ref()
-                                .or(modules.ws.as_ref())
-                                .map(RpcRequestMetrics::same_port)
-                                .unwrap_or_default(),
-                        ),
-                    )
+                    .set_rpc_middleware(self.rpc_middleware)
```

Comment:
this now puts the metrics setup on the caller, we don't want this.

can we please keep the rpc_middleware setup as it was before and then integrate the custom middleware in a next step
there's a lot that can go wrong here